src/lib/components/MyReservations.svelte

```diff
@@ -117,50 +102,56 @@
 		});
 	};
 
-	const groupRsvs = (resPeriod, allRsvs, userPastRsvs) => {
-		let rsvs;
+	type ReservationsByMonth = {
+		month?: string;
+		rsvs: ReservationData[];
+	};
+
+	const groupRsvs = (
+		resPeriod: ReservationPeriod,
+		allRsvs: ReservationData[],
+		userPastRsvs: ReservationData[]
+	): ReservationsByMonth[] => {
+		let rsvs: ReservationData[] = [];
 		if (resPeriod === 'upcoming') {
 			rsvs = allRsvs.filter((rsv) => {
-				return rsv.user.id === $user.id && getResPeriod(rsv) === resPeriod;
+				return rsv?.user?.id === $user.id && getResPeriod(rsv) === resPeriod;
 			});
 		} else if (resPeriod === 'past') {
 			rsvs = userPastRsvs.filter((rsv) => getResPeriod(rsv) === resPeriod);
 		}
 
-		let sorted = sortChronologically(rsvs, resPeriod);
+		const sorted = sortChronologically(rsvs, resPeriod);
 
 		if (resPeriod === 'upcoming') {
 			return [{ rsvs: sorted }];
-		} else if (resPeriod === 'past') {
-			let curM;
+		} else {
 			return sorted.reduce((grps, rsv) => {
-				let m = monthIdxFromDateStr(rsv.date);
-				if (m === curM) {
-					grps[grps.length - 1].rsvs.push(rsv);
+				const month = dayjs(rsv.date).format('MMMM');
```

Comment:
@nrxszvo if we can eliminate more of the custom utility functions implemented this would save us code.